### DISCUSSION TITLE
Authentication with new createBrowserRouter failed with Context API, but working fine with BrowserRouter

### DISCUSSION THREAD

**Discussion by shrijan00003** [remix-run/react-router#9927] [Q&A]
I have tried to implement authentication with the new `createBrowserRouter` API with context API, which seems to have weird behaviors, not sure whether the issue is with the implementation or with API. 

Please check both examples:
Working example with BrowserRouter: https://stackblitz.com/edit/react-ts-jiaksh?file=App.tsx
(Not working) Example with `createBrowserRouter` : https://stackblitz.com/edit/react-ts-qjrtdp?file=App.tsx

**Comment by shrijan00003** [comment]
@whtjs @Vanilla2 Any suggestions? Is this the bug or I am missing something?

**Comment by jcarballido** [comment]
Struggling with something similar. Changing the custom context in the Auth Provider is not passing to other routes, only the one route that changes it  (e.g., '/login'). The upated context only propogates if I navigate using the UI, but if I try to navigate by changing the URI directly, the custom context reverts to its original state...
